Info-Orbs/src/widgets/stockScreenWidget.cpp

```diff
@@ -0,0 +1,113 @@
+#include "widgets/stockScreenWidget.h"
+#include <config.h>
+#include <sstream>
+#include <iomanip>
+#include <HTTPClient.h>
+#include <ArduinoJson.h>
+
+StockScreenWidget::StockScreenWidget(ScreenManager &manager) : Widget(manager), currentStockIndex(0), m_stockDelay(6000) {
+}
+
+void StockScreenWidget::setup() {
+    m_stockDelayPrev = 0;
+    stockTickers = {"ABNB", "BRCC", "GOOG", "TSLA", "GME"};
+}
+
+void StockScreenWidget::draw() {
+    // Get initial stock then every m_stockDelay milliseconds
+    if (m_stockDelayPrev == 0 || (millis() - m_stockDelayPrev) >= m_stockDelay) {
+        getStockData();
+        displayStock(currentStockIndex, TFT_WHITE, TFT_BLACK);
+        m_stockDelayPrev = millis();
+        currentStockIndex = (currentStockIndex + 1) % stockTickers.size();
+    }
+}
+
+void StockScreenWidget::update() {
```

Comment:
This will need to be updated to use the new `force` bool that was just merged. At least to implement the new method signature.

Reply:
I saw you all were chatting about this. I am fine with whatever you all decide. I would just have to see how to do it.

Reply:
look at the new version of Widget and either WeatherWidget or ClockWidget.